src/vm/database/structures.rs

```diff
@@ -93,3 +93,114 @@ clarity_serializable!(i128);
 clarity_serializable!(u128);
 clarity_serializable!(u64);
 clarity_serializable!(Contract);
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct STXBalance {
+    pub amount_unlocked: u128,
+    pub amount_locked: u128,
+    pub unlock_height: u64,
+}
+clarity_serializable!(STXBalance);
+
+#[derive(Debug)]
+pub enum STXBalanceError {
```

Comment:
Yeah I thought about this, I ended up optimizing for modularity by adding a custom error type, so that the caller can panic (or not) with a specific and contextual message.

Reply:
All instances of these errors are either converted into `.unwrap()` or `.expect()` by their callers, or converted to an `ArthithmeticError` without consideration of the error type.  So, I don't buy the argument for introducing this type if it's not going to be used in a meaningful way.  But, if you insist on keeping it, then it should not be `pub` until there is at least one other module using it.

Reply:
Making this enum private is not an option, since
```
pub fn debit(&mut self, amount: u128, block_height: u64) -> Result<()>
pub fn credit(&mut self, amount: u128, block_height: u64) -> Result<()>
pub fn transfer_to(&mut self, recipient: &mut STXBalance, amount: u128, block_height: u64) -> Result<()>
```
would end up leaking a private type, leading to a compiler error.
We could return an optional instead which looks like a downgrade.